webapp/views/App/views/Designer/TaxonomyList/Row/store/actions/useEdit.js

```diff
@@ -1,29 +1,28 @@
 import { useCallback } from 'react'
-import { useDispatch } from 'react-redux'
 import { matchPath, useHistory, useLocation } from 'react-router'
+import { useDispatch } from 'react-redux'
 
-import * as ObjectUtils from '@core/objectUtils'
 import { appModuleUri, designerModules } from '@webapp/app/appModules'
 
-import * as TaxonomyActions from '@webapp/views/App/views/Designer/TaxonomyDetail/actions'
-
 import * as Taxonomy from '@core/survey/taxonomy'
+import * as TaxonomyActions from '@webapp/views/App/views/Designer/TaxonomyDetail/actions'
 
 import { State } from '../state'
 
 export const useEdit = () => {
-  const dispatch = useDispatch()
   const history = useHistory()
-
+  const dispatch = useDispatch()
   const { pathname } = useLocation()
   const inTaxonomiesPath = Boolean(matchPath(pathname, appModuleUri(designerModules.taxonomies)))
 
   return useCallback(({ state }) => {
     const taxonomy = State.getTaxonomy(state)
+    const taxonomyUuid = Taxonomy.getUuid(taxonomy)
     if (!inTaxonomiesPath) {
-      dispatch(TaxonomyActions.setTaxonomyForEdit(Taxonomy.getUuid(taxonomy)))
+      dispatch(TaxonomyActions.setTaxonomyForEdit(taxonomyUuid))
+      State.getOnOpenTaxonomy(state)(taxonomyUuid)
```

Comment:
is this a mistake? the order seems wrong .
shouldn't it be ```State.getOnOpenTaxonomy(taxonomyUuid)(state)```

Reply:
yes!

Reply:
not really, State.getOnOpenTaxonomy(state) is a callback that is passed with props to this component, should I pass the callback with the params of the function instead of store it into the state(which is not normal)?

Reply:
right!